src/lib/Components/Bidding/Context/TimeOffsetProvider.tsx

```diff
@@ -0,0 +1,94 @@
+import * as PropTypes from "prop-types"
+import * as React from "react"
+import { metaphysics } from "../../../metaphysics"
+
+interface SystemResults {
+  data: {
+    system: {
+      time: {
+        unix: number
+      }
+    }
+  }
+}
+
+const getLocalTimestampInMilliSeconds = () => {
+  return Date.now()
+}
+
+const fetchSystemTime = () => {
+  return metaphysics<SystemResults>({
+    query: `
+      {
+        system {
+          time {
+            unix
+          }
+        }
+      }
+    `,
+  })
+}
+
+const getGravityTimestampInMilliSeconds = async () => {
+  const startTime = getLocalTimestampInMilliSeconds()
+  const { data } = await fetchSystemTime()
+
+  const possibleNetworkLatencyInMilliSeconds = (getLocalTimestampInMilliSeconds() - startTime) / 2
+  const serverTimestampInMilliSeconds = data.system.time.unix * 1e3 + possibleNetworkLatencyInMilliSeconds
+
+  if (__DEV__ && typeof jest === "undefined") {
```

Comment:
I don’t know how `__DEV__` code is usually stripped from production builds, but maybe it looks just for `if (__DEV__) { … }`, in which case you may want to nest the `if (typeof jest === "undefined") { … }` part.

Reply:
Yeah I thought about it too. Let's make it safer by nesting.